features/mbedtls/targets/TARGET_NUVOTON/TARGET_M480/ecp/ecp_internal_alt.c

```diff
@@ -0,0 +1,835 @@
+
+#define MODOP_DIV           (0x0UL << CRPT_ECC_CTL_MODOP_Pos)
+#define MODOP_MUL           (0x1UL << CRPT_ECC_CTL_MODOP_Pos)
+#define MODOP_ADD           (0x2UL << CRPT_ECC_CTL_MODOP_Pos)
+#define MODOP_SUB           (0x3UL << CRPT_ECC_CTL_MODOP_Pos)
+
+/**
+ * \brief           Check if MPI has been normalized
+ *
+ * \param N         Input MPI which is to check
+ * \param P         Prime modulus
+ *
+ * \return          0 if not normalized,
+ *                  1 if normalized
+ */
+#define MBEDTLS_INTERNAL_MPI_IS_NORM(N, P)                          \
+    ((mbedtls_mpi_cmp_int(&N, 0) >= 0) && (mbedtls_mpi_cmp_mpi(&N, &P) < 0))
+  
+        
+/**
+ * \brief           Normalize MPI if it is not normalized yet
+ *
+ * \param R         Holds pointer to normalized MPI (N1 or N2)
+ * \param N1        Input MPI which is to normalize
+ * \param N2        Output MPI which holds normalized N1 if N1 is not normalized yet
+ * \param P         Prime modulus
+ */
+#define MBEDTLS_INTERNAL_MPI_NORM(R, N1, N2, P)                     \
+    do {                                                            \
+        if (MBEDTLS_INTERNAL_MPI_IS_NORM(N1, P)) {                  \
+            *R = &N1;                                               \
+        } else {                                                    \
+            MBEDTLS_MPI_CHK(mbedtls_mpi_mod_mpi(&N2, &N1, &P));     \
+            *R = &N2;                                               \
+        }                                                           \
+    } while(0)
+
+/**
+ * \brief           Configure ECCOP operation, start it, and wait for its completion
+ *
+ * \param grp       ECP group
+ * \param R         Destination point
+ * \param m         Integer by which to multiply P
+ * \param P         Point to multiply by m
+ * \param n         Integer by which to multiply Q
+ * \param Q         Point to be multiplied by n
+ * \param eccop     ECCOP code. Could be ECCOP_POINT_MUL/ADD/DOUBLE
+ *                  Dependent on passed-in eccop, only partial parameters among m/P/n/Q are needed and checked.
+ *                  ECCOP_POINT_MUL     R = m*P
+ *                  ECCOP_POINT_ADD     R = P + Q
+ *                  ECCOP_POINT_DOUBLE  R = 2*P
+ *
+ * \return          0 if successful
+ *
+ * \note            P/Q must be normalized (= affine). R would be normalized.
+ *
+ * \note            m/n could be negative.
+ *
+ * \note            ECC accelerator doesn't support R = 0, and we need to detect it additionally.
+ *                  For R = P + Q or R = 2*P, we can detect all R = 0 cases.
+ *                  For R = m*P, we can detect all R = 0 cases only if grp->N (order) is a prime.
+ *
+ * \note            According to ECCOP operation, n is unnecessary. But to be consistent with R = m*P + n*Q,
+ *                  n is kept with unused modifier.
+ *                  
+ */
+int mbedtls_internal_run_eccop(const mbedtls_ecp_group *grp,
```

Comment:
If it's meant to an internal function, please make it `static` and remove the `mbedtls_` prefix.

Reply:
@hanno-arm I remove the `mbedtls_` prefix in https://github.com/ARMmbed/mbed-os/pull/5812/commits/df76e297b44dc346fb80e7d140cbf140a68254a9 for internal functions. For Nuvoton's self-test on ECP alter. module, I add a macro `NU_CRYPTO_SELF_TEST` to conditionally remove the `static` modifier in https://github.com/ARMmbed/mbed-os/pull/5812/commits/25253529aee9f0d9f29e13c68d69ce58dea3af1d.